tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -155,6 +167,8 @@ object NativeConfig {
       targetTriple = None,
       gc = GC.default,
       lto = LTO.default,
+      maxInlineDepth = 1000,
```

Comment:
I think this default value is a bit to big. If we would like allow to infinite number of inlines, then probably on `Option[Int]` would be a better choice. `None` would then mean unlimited